mkt/search/filters.py

```diff
@@ -47,16 +47,23 @@ def filter_queryset(self, request, queryset, view):
             rules.append(
                 (query.Fuzzy, {'value': q, 'boost': 2, 'prefix_length': 1}))
 
+        # Apply rules to search on few base fields. Some might not be present
+        # in every document type / indexes.
         for k, v in rules:
-            for field in ('name', 'app_slug', 'author'):
+            for field in ('name', 'title', 'short_title',
+                          'app_slug', 'author', 'url'):
                 should.append(k(**{field: v}))
 
         # Exact matches need to be queried against a non-analyzed field. Let's
-        # do a term query on `name.raw` for an exact match against the app
+        # do a term query on `name.raw` for an exact match against the item
         # name and give it a good boost since this is likely what the user
         # wants.
         should.append(query.Term(**{'name.raw': {'value': q, 'boost': 10}}))
 
+        # FIXME: for websites, what should we match against ? "title" is going
+        # to be extracted from websites <title> so it won't work. "short_title"
+        # is only intended for display purposes. Or should we re-purpose it ?
```

Comment:
For instance, the `title` for nytimes.com would be `"The New York Times - Breaking News, World News & Multimedia"` and not just `"The New York Times"` - Users are going to use the latter, not the former.

(Another example that is inversed: the `title` for theguardian.com would be `"Latest news, sport and comment from the Guardian | The Guardian"` when most users would search for `"Guardian"` or `"The Guardian"` so the exact match will never work if we do it against the full title.

At the same time I feel that the short title would not work either: for nytimes.com it might be `"NYTytimes"` and not "The New York Times".